src/request.ts

```diff
@@ -0,0 +1,126 @@
+import type { DocumentTypeDecoration } from "@graphql-typed-document-node/core";
+import { createSha256, extractOperationName, pruneObject } from "helpers";
+
+export type ModeFlags = "persisted" | "document" | "both";
+
+export type DocumentIdFn = <TResult, TVariables>(
+	query: DocumentTypeDecoration<TResult, TVariables>,
+) => string | undefined;
+
+export type GraphQLRequest<TVariables> = {
```

Comment:
Should look at making this a discriminated union where documentId is required when choosing a persisted query mode. Mode is a nice discriminator for it anyways

Reply:
I had that initially, but since we support 3 modes (document, persistent, both) this ended up being simplest